cmake/SndFileConfig.cmake.in

```diff
@@ -5,19 +5,24 @@ set(SndFile_VERSION_PATCH @PROJECT_VERSION_PATCH@)
 
 set (SndFile_WITH_EXTERNAL_LIBS @SndFile_WITH_EXTERNAL_LIBS@)
 set (SndFile_WITH_MPEG @SndFile_WITH_MPEG@)
+set (SndFile_STATIC NOT @BUILD_SHARED_LIBS@)
 
 @PACKAGE_INIT@
 
 include (CMakeFindDependencyMacro)
 
-if (SndFile_WITH_EXTERNAL_LIBS AND NOT @BUILD_SHARED_LIBS@)
+if ((SndFile_WITH_EXTERNAL_LIBS OR SndFile_WITH_MPEG) AND SndFile_STATIC)
+    list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
```

Comment:
The side-effect of this call should be cleaned up after the calls to `find_dependency`, with:
```cmake
if((SndFile_WITH_EXTERNAL_LIBS OR SndFile_WITH_MPEG) AND NOT SndFile_IS_SHARED)
  list(REMOVE_ITEM CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
endif()
```